Apromore-Commons/src/main/java/org/apromore/commons/io/FilenameUtils.java

```diff
@@ -0,0 +1,122 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2020 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+
+package org.apromore.commons.io;
+
+import org.apromore.commons.io.Constants;
+
+import java.lang.Math;
+import java.util.regex.Pattern;
+import java.util.Collections;
+import java.util.List;
+/**
+ * Various filename utility functions
+ */
+public final class FilenameUtils {
+
+    private static final String DELIMITERS = "\\._\\+\\- ";
+    private static final String MERGED = "_merged";
+    private static final int MAX_LENGTH = Constants.VALID_FILENAME_MAX_LENGTH - MERGED.length();
```

Comment:
Suggest renaming the package name from "org.apromore.commons.io" to "org.apromore.commons.item", and the class name from FilenameUtils to ItemNameUtils. Apromore has been using "item" for process models, logs (e.g. SummaryItemRenderer in Apromore-Portal) and "Item" seems to be OK and a clean name to be used for these things, but "File" is not OK as it would be confusing with the "File" terminology used for computer files, directories as described in Apache-Commons' FilenameUtils javadocs (https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/FilenameUtils.html), and IO is of course a standard term used for computer files, streams, etc. as described in apache.commons.io package javadocs (http://commons.apache.org/proper/commons-io/apidocs/index.html?org/apache/commons/io/package-summary.html). We should not mix up these two notions: "Item" in Apromore and the standard term "File" for computer files.